src/recovery/recover_entity.rs

```diff
@@ -0,0 +1,1021 @@
+use std::{hash::Hash, ops::Add, sync::RwLock};
+
+use crate::prelude::*;
+
+impl Profile {
+    async fn new_entity<E: IsEntity + std::fmt::Debug + std::hash::Hash + Eq>(
```

Comment:
This is a WEIRD impl, using gateway, which it should not, it should take a `pre_derive_keys_cache` instead! and use the "next key"! but I have not implemented that....

relevant thread: https://rdxworks.slack.com/archives/C06EBEA0SGY/p1726135124592619?thread_ts=1726133029.389349&cid=C06EBEA0SGY